src/gt4py/next/errors/excepthook.py

```diff
@@ -21,7 +21,8 @@
 """
 
 import sys
-from typing import Callable
+import types
+from typing import Callable, Optional
```

Comment:
We should use `collections.abc.Callable` since Python 3.9 but maybe we can skip it for now until we fully deprecate 3.8 and 3.9 support...